src/contracts/interfaces/IDelegationManager.sol

```diff
@@ -1,81 +1,249 @@
 // SPDX-License-Identifier: BUSL-1.1
 pragma solidity =0.8.12;
 
-import "./IDelegationTerms.sol";
+import "./IStrategy.sol";
 
 /**
  * @title The interface for the primary delegation contract for EigenLayer.
  * @author Layr Labs, Inc.
  * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
  * @notice  This is the contract for delegation in EigenLayer. The main functionalities of this contract are
  * - enabling anyone to register as an operator in EigenLayer
- * - allowing new operators to provide a DelegationTerms-type contract, which may mediate their interactions with stakers who delegate to them
- * - enabling any staker to delegate its stake to the operator of its choice
- * - enabling a staker to undelegate its assets from an operator (performed as part of the withdrawal process, initiated through the StrategyManager)
+ * - allowing operators to specify parameters related to stakers who delegate to them
+ * - enabling any staker to delegate its stake to the operator of its choice (a given staker can only delegate to a single operator at a time)
+ * - enabling a staker to undelegate its assets from the operator it is delegated to (performed as part of the withdrawal process, initiated through the StrategyManager)
  */
 interface IDelegationManager {
+    // @notice Struct used for storing information about a single operator who has registered with EigenLayer
+    struct OperatorDetails {
```

Comment:
can we add a metadata url to this. this will be where the operator's static metadata is hosted. we don't need to store this, we only need to index it. wdyt?

Reply:
If we aren't storing it then doesn't it belong in an event rather than this struct used specifically for storage?

Thinking something like `OperatorMetadataURIUpdated(address indexed operator, bytes metadataURI)`

If we want it in storage then it could make sense here, but I do think it depends on how often we'd expect it to be updated (like maybe operators want more frequent updates to their metadata URI than these other addresses? right now there is a single setter function)

Reply:
metadataURI being a string is nicer, but other than that it works.

we discussed that adding a `string metadataURI` to the register operator and a `modifyMetadataURI` function

Reply:
ah yeah, made it a string. just pushed relevant commits; please take a look and resolve this if you think it all will work.

Reply:
works